### ISSUE TITLE
Unable to switch to nvidia driver

### ISSUE THREAD

**Issue by slade991** [Askannz/optimus-manager#102]
**Describe the bug**

When i try to switch to nvidia, i get logged out, the welcome screen of my computer show up (asus) and stay there. I can switch to another TTY. If i switch back to TTY1 le welcome screen is gone and replaced by a blinking underscore. But nothing happens.

**System info**
Please include :
- Your distribution
- Your desktop manager (KDE, GNOME, Cinnamon...), if you have one
- You display manager (SDDM, GDM, LightDM...), if you have one. If you do not know, you are probably using the one coming with your desktop manager.
- Your laptop model, if you think it could be related to the problem

Manjaro latest, with GNOME. Geforce GTX 1050 Ti.

**Logs**
If you are able to, please attach all the logs you can find in `/var/log/optimus-manager/`. *Do not directly copy paste the logs into the GitHub issue.* Join them as text files or save them to pastebins (https://pastebin.com/) and post the links here.

Xorg doesn't output any logs. And the optimus logs look completely normal.
 
boot_setup.log -> [https://termbin.com/dk7t](https://termbin.com/dk7t)
gpu_setup.log -> [https://termbin.com/xkp2](https://termbin.com/xkp2)
prime_setup.log -> [https://termbin.com/7f6a](https://termbin.com/7f6a)

There is a lot of trial and error here. I tried pretty everything i could read here and there, DRI, modeset, bbswitch/ nouveau /none, kernel 4.19, 5.1. I always have the same result. Intel boot correctly but not nvidia.

Here is an output of dmesg meanwhile on two different failed nvidia startup:
![IMG20190709231549](https://user-images.githubusercontent.com/10143703/60907974-047b3e00-a2a5-11e9-8c34-94ea7b7765f8.jpg)
 
![IMG20190709233005](https://user-images.githubusercontent.com/10143703/60907991-0ba24c00-a2a5-11e9-804a-b113fc57aabc.jpg)

thanks for the help :)

**Comment by Socterean** [comment]
It's because of this [issue](https://github.com/Askannz/optimus-manager/issues/68).

TLDR:
Manjaro uses a custom `sddm.conf` in `/etc/sddm.conf` and to make **optimus-manager** work you have to edit `/etc/sddm.conf` and comment by inserting `#` in front of **DisplayCommand** and **DisplayStopCommand** lines.

After that it works.

Also, I believe it's a good ideea for the creator to include this in the installation Wiki of `optimus-manager`.

**Comment by slade991** [comment]
Hey thanks for the input,
i do not have a sddm.conf, although i have a /etc/sddm.conf.d/20-optimus-manager.conf
with this content:
[X11]
DisplayCommand=/sbin/prime-offload
DisplayStopCommand=/sbin/prime-switch

Should i comment it?

**Comment by Askannz** [comment]
@Socterean That's an issue with Manjaro KDE, doesn't apply to GNOME.

@slade991 I can see from your logs that you have added a custom configuration to `/etc/optimus-manager/xorg-nvidia.conf`  :

```
Loaded extra Nvidia Xorg options (2 lines)
```

Try removing those lines. If that doesn't work, get a log from the display manager (`journalctl -u gdm`).

**Comment by slade991** [comment]
@Askannz  i added to try to make it work it was not working without it.

The only 2 lines in there are those:

BusID "PCI:02:00.0"
Driver         "nvidia"

**Comment by Askannz** [comment]
Does it work if you boot directly into Nvidia mode with `--set-startup nvidia` ?

**Comment by jorpilo** [comment]
I was getting a very similar problem.
On Manjaro KDE after switching to Nvidia the latop frozen.

After installing bbswitch, setting bbswitch as backend and restarting the PC now I am able to make optimus-manager work perfectly. If you don't restart the PC after changing the backend and just try to change to Nvidia, it still freezes.

I have installed ONLY the nvidia drivers using the Manjaro hardware utility. Maybe I should also have linux-driver too

**Comment by slade991** [comment]
@Askannz my computer is now at the Asus service center for Wi-Fi issue do I won't be able to check for quite a bit. 

@jorpilo I did this already as I mentionned and I did restart quite a lot of time as well.

**Comment by Askannz** [comment]
Do you still have the issue with the latest version of optimus-manager ?

**Comment by slade991** [comment]
Hello,
thanks for the follow up, it turned out i had a issue with my motherboard, which got replaced. I do not know if that was related to the issue i had with optimus-manager as i am now using pop_os with it's own graphic card switcher (not sure what they use).
So unfortunately i don't know :)
I'll close it for now though, thank you for your help.
